qlty-cli/src/commands/coverage/complete.rs

```diff
@@ -0,0 +1,403 @@
+        eprintln_unless!(self.quiet, "qlty {}", LONG_VERSION.as_str());
+        eprintln_unless!(self.quiet, "{}", style("https://qlty.sh/d/coverage").dim());
+        eprintln_unless!(self.quiet, "");
+    }
+
+    fn print_section_header(&self, title: &str) {
+        eprintln_unless!(self.quiet, "{}", style(title).bold().reverse());
+        eprintln_unless!(self.quiet, "");
+    }
+
+    fn print_settings(&self, _settings: &Settings) {
+        self.print_section_header(" SETTINGS ");
+
+        eprintln_unless!(
+            self.quiet,
+            "    cwd: {}",
+            std::env::current_dir()
+                .unwrap_or(PathBuf::from("ERROR"))
+                .to_string_lossy()
+        );
+
+        if let Some(tag) = &self.tag {
+            eprintln_unless!(self.quiet, "    tag: {}", tag);
+        }
+        if let Some(override_branch) = &self.override_branch {
+            eprintln_unless!(self.quiet, "    override-branch: {}", override_branch);
+        }
+        if let Some(override_commit_sha) = &self.override_commit_sha {
+            eprintln_unless!(
+                self.quiet,
+                "    override-commit-sha: {}",
+                override_commit_sha
+            );
+        }
+        if let Some(override_pr_number) = &self.override_pr_number {
+            eprintln_unless!(self.quiet, "    override-pr-number: {}", override_pr_number);
+        }
+        if let Some(project) = &self.project {
+            eprintln_unless!(self.quiet, "    project: {}", project);
+        }
+
+        eprintln_unless!(self.quiet, "");
+    }
+
+    fn print_metadata(&self, plan: &Plan) {
+        self.print_section_header(" METADATA ");
+        if !plan.metadata.ci.is_empty() {
+            eprintln_unless!(self.quiet, "    CI: {}", plan.metadata.ci);
+        }
+
+        eprintln_unless!(self.quiet, "    Commit: {}", plan.metadata.commit_sha);
+        if !plan.metadata.pull_request_number.is_empty() {
+            eprintln_unless!(
+                self.quiet,
+                "    Pull Request: #{}",
+                plan.metadata.pull_request_number
+            );
+        }
+
+        if !plan.metadata.branch.is_empty() {
+            eprintln_unless!(self.quiet, "    Branch: {}", plan.metadata.branch);
+        }
+
+        if !plan.metadata.build_id.is_empty() {
+            eprintln_unless!(self.quiet, "    Build ID: {}", plan.metadata.build_id);
+        }
+
+        eprintln_unless!(self.quiet, "");
+    }
+
+    fn print_authentication_info(&self, token: &str) {
+        let token_type = if token.starts_with(COVERAGE_TOKEN_WORKSPACE_PREFIX) {
+            "Workspace Token"
+        } else if token.starts_with(COVERAGE_TOKEN_PROJECT_PREFIX) {
+            "Project Token"
+        } else if let Ok(oidc_regex) = Regex::new(OIDC_REGEX) {
+            if oidc_regex.is_match(token) {
+                "OIDC"
+            } else {
+                "Unknown"
+            }
+        } else {
+            "ERROR"
```

Comment:
This level of complexity is about where I start to get concerned about duplication